presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -382,6 +396,76 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
         }
     }
 
+    @Override
+    public ListenableFuture<?> removeRemoteSource(TaskId remoteSourceTaskId)
+    {
+        URI remoteSourceUri = uriBuilderFrom(taskLocation)
+                .appendPath("remote-source")
+                .appendPath(remoteSourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
+                taskId,
+                remoteSourceUri,
+                maxErrorDuration,
+                errorScheduledExecutor,
+                "Remove exchange remote source");
+
+        SettableFuture<?> future = SettableFuture.create();
+        removeRemoteSourceHelper(errorTracker, request, future);
+        return future;
+    }
+
+    private void removeRemoteSourceHelper(RequestErrorTracker errorTracker, Request request, SettableFuture<?> future)
```

Comment:
This method name sounds more like it removes something called remoteSourceHelper. How about `handleRemoteSourceRemoval`?

Reply:
or `doRemoveRemoteSource`  ? ;)